Source/Blazorise/Utilities/ReflectionHelper.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+
+namespace Blazorise.Utilities;
+
+/// <summary>
+/// An helper for reflection based apis.
+/// </summary>
+public class ReflectionHelper
+{
+    /// <summary>
+    /// Gets the public instance properties of the specified type.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <returns></returns>
+    public static PropertyInfo[] GetPublicProperties<T>()
+    {
+        var properties = typeof( T ).GetProperties( BindingFlags.Public | BindingFlags.Instance );
+        return properties;
+    }
+
+    /// <summary>
+    /// Based on this particular property, tries to resolve a caption out of the attributes.
+    /// </summary>
+    /// <param name="propertyInfo"></param>
+    /// <returns></returns>
+    public static string ResolveCaption( PropertyInfo propertyInfo )
+    {
+        var displayAttribute = propertyInfo.GetCustomAttribute<DisplayAttribute>();
+        if ( displayAttribute is not null )
+        {
+            return displayAttribute.GetName();
+        }
+
+        var displayNameAttribute = propertyInfo.GetCustomAttribute<DisplayNameAttribute>();
+        if ( displayNameAttribute is not null )
+        {
+            return displayNameAttribute.DisplayName;
+        }
+
+        return propertyInfo.Name;
```

Comment:
Oh now I see that you return property name as a default. That leaves it only to properly format Title cased name to include spaces as a separator.

Reply:
Right, see related comment above. Us trying to be smart about it, relies on people using the PascalCase convention and not being able to opt out if we make it default.

Reply:
The default naming convention in C# is PascalCase, according to the [docs](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions). And people can always use `DisplayAttribute` if they don't like it.